ArchUnitNET/Domain/PlantUml/Export/PlantUmlFileBuilder.cs

```diff
@@ -95,57 +99,311 @@ public PlantUmlFileBuilder WithDependenciesFrom(IEnumerable<IType> types, Genera
                 generationOptions.IncludeNodesWithoutDependencies
                     ? nodes.Values
                     : nodes.Where(node =>
-                            dependencies.SelectMany(dep => new[] {dep.Origin, dep.Target}).Contains(node.Key.FullName))
+                            _dependencies.SelectMany(dep => new[] {dep.Origin, dep.Target}).Contains(node.Key.FullName))
                         .Select(node => node.Value);
 
             _diagram.AddElements(nodeElements);
-            _diagram.AddElements(dependencies);
+            _diagram.AddElements(_dependencies);
 
             return this;
         }
 
-        public PlantUmlFileBuilder WithDependenciesFrom(IEnumerable<Slice> slices, GenerationOptions generationOptions = null)
+        public PlantUmlFileBuilder WithDependenciesFrom(IEnumerable<Slice> slices)
         {
-            var sliceList = slices.Distinct().ToList();
-            if (generationOptions == null)
+            _sliceList = slices.Distinct().ToList();
+            RemovePatternInappropriateSlices();
+
+            var nodes = new Dictionary<Slice, IPlantUmlElement>();
+            var generationOptions = new GenerationOptions();
```

Comment:
`generationOptions` is pretty useless if it can't be passed as an option to this function. I think it should be added back to the parameters, but otherwise it can be removed completely.